pkg/cpio/newc/newc.go

```diff
@@ -117,6 +117,9 @@ func (w *writer) WriteRecord(f cpio.Record) error {
 
 	buf := &bytes.Buffer{}
 	hdr := headerFromInfo(f.Info)
+	if f.ReadCloser == nil {
```

Comment:
GNU cpio sets the size to 0, for newc, and if the size is non-zero, on read, both it and the kernel collapse in a heap. This change makes the newc format we produce compatible with that code. It is not clear at all that ALL formats require a zero size, hence my reluctance to push that decision up to cpio from newc. But this change is a result of that discovery: if we don't set size to 0, then the two cpio readers I have (gnu and linux kernel) can't read it.

The docs on cpio are skimpy; it was dropped from POSIX in 2001, so in this case, code rules.